### DISCUSSION TITLE
Board versioning

### DISCUSSION THREAD

**Discussion by gideonmaina** [CodeForAfrica/sensors.AFRICA-AQ-sensors-hardware-GSM#7] [Workflow]
Let us come up with a list of naming conventions for our board. *_Not limited to GSM enabled boards _*.
@augustinemwendwa72 @usmanbiu . We can move this to discussions.

**Comment by augustinemwendwa72** [comment]
@gideonmaina @usmanbiu We can come up with a uniform convention for naming both hardware and software versions. Also, a readme file or any form of document

For the versioning, we can have:

- MCU Board type (eg, E32 for ESP32)
- Version year and month (eg, 0824)
- Group (H, S, or E) for Hardware, Software or Enclosure.
- No need to include all the details, the aim is to have a Unique version. An example name would look like E320824H. E320824H and E320824S should be Compatible.

We can even have Nicknames for each version, something easily memorable.

**Reply by gideonmaina** [reply]
> @gideonmaina @usmanbiu We can come up with a uniform convention for naming both hardware and software versions. Also, a readme file or any form of document
> 
> For the versioning, we can have:
> 
> * MCU Board type (eg, E32 for ESP32)
> * Version year and month (eg, 0824)
> * Group (H, S, or E) for Hardware, Software or Enclosure.
> * No need to include all the details, the aim is to have a Unique version. An example name would look like E320824H. E320824H and E320824S should be Compatible.
> 
> We can even have Nicknames for each version, something easily memorable.

@usmanbiu @augustinemwendwa72 We can start with the GROUP e.g for hardware `HW` then MCU  followed by month-year.  Any unique tag for alpha/beta, nickname or even modem type can be concatenated by an underscore.

For Firmware we can have a different naming convention.

Both can have a `versions.md` to show the change logs

**Comment by usmanbiu** [comment]
@augustinemwendwa72 this makes a lot of sense. We should add a tag to let us know the boards with a simcom or quecktel chip. I recommend we add "(S)" at the end of the names of the boards with the simcom chip e.g E820819(S), while leaving the ones with a quecktel chip with no tag. This is because we only have one version of the board with a simcom chip while we are already working on the third version with a quecktel chip. 

I also think it would be a good idea to give the nicknames for ease of use during conversations e.g kawari, yula etc

cc @gideonmaina 

> @gideonmaina @usmanbiu We can come up with a uniform convention for naming both hardware and software versions. Also, a readme file or any form of document
> 
> For the versioning, we can have:
> 
>     * MCU Board type (eg, E32 for ESP32)
> 
>     * Version year and month (eg, 0824)
> 
>     * Group (H, S, or E) for Hardware, Software or Enclosure.
> 
>     * No need to include all the details, the aim is to have a Unique version. An example name would look like E320824H. E320824H and E320824S should be Compatible.
> 
> 
> We can even have Nicknames for each version, something easily memorable.

**Comment by gideonmaina** [comment]
@augustinemwendwa72  and @usmanbiu  moving this to a discussion.
